llvm/lib/Analysis/ScalarEvolution.cpp

```diff
@@ -8494,9 +8492,8 @@ void ScalarEvolution::forgetLoop(const Loop *L) {
         ++I;
     }
 
-    auto LoopUsersItr = LoopUsers.find(CurrL);
-    if (LoopUsersItr != LoopUsers.end())
-      llvm::append_range(ToForget, LoopUsersItr->second);
+    auto Users = LoopUsers.lookup(CurrL);
```

Comment:
This will copy the vector. Many similar issues below.

Reply:
Ah yes, sorry about missing this subtle point. Fixed now.